src/producer_processor_saver/Main.java

```diff
@@ -0,0 +1,43 @@
+package producer_processor_saver;
+
+public class Main {
+
+    public static void main(String[] args) {
+        Logic logic = new Logic(100, 5, 5, 1000, 5000, 1000);
+
+        Thread loader = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                logic.loadAll();
+            }
+        });
```

Comment:
Java had lambdas for like 5 years - this could be a one liner - `new Thread(()->logic.LoadAll())`